File: content/project/single-module-application/src/main/java/${project_rootPath}/screen/main/MainScreen.java
package ${project_rootPackage}.screen.main;

import io.jmix.ui.ScreenTools;
import io.jmix.ui.component.AppWorkArea;
import io.jmix.ui.component.Button;
import io.jmix.ui.component.Window;
import io.jmix.ui.component.mainwindow.Drawer;
import io.jmix.ui.icon.JmixIcon;
import io.jmix.ui.navigation.Route;
import io.jmix.ui.screen.Screen;
import io.jmix.ui.screen.Subscribe;
import io.jmix.ui.screen.UiController;
import io.jmix.ui.screen.UiControllerUtils;
import io.jmix.ui.screen.UiDescriptor;
import org.springframework.beans.factory.annotation.Autowired;

@UiController("${normalizedPrefix_underscore}MainScreen")
@UiDescriptor("main-screen.xml")
@Route(path = "main", root = true)
public class MainScreen extends Screen implements Window.HasWorkArea {

    @Autowired
    private ScreenTools screenTools;

    @Autowired
    private AppWorkArea workArea;
    @Autowired
    private Drawer drawer;
    @Autowired
    private Button collapseDrawerButton;


    @Override
    public AppWorkArea getWorkArea() {
        return workArea;
    }

    @Subscribe("collapseDrawerButton")
    private void onCollapseDrawerButtonClick(Button.ClickEvent event) {
        drawer.toggle();
        if (drawer.isCollapsed()) {
            collapseDrawerButton.setIconFromSet(JmixIcon.CHEVRON_RIGHT);
        } else {
            collapseDrawerButton.setIconFromSet(JmixIcon.CHEVRON_LEFT);
        }
    }

    @Subscribe
    public void onAfterShow(AfterShowEvent event) {
        screenTools.openDefaultScreen(
                UiControllerUtils.getScreenContext(this).getScreens());

        screenTools.handleRedirect();
    }
}

File: content/project/single-module-application-kotlin/src/main/kotlin/${project_rootPath}/screen/main/MainScreen.kt
package ${project_rootPackage}.screen.main

import io.jmix.ui.ScreenTools
import io.jmix.ui.component.AppWorkArea
import io.jmix.ui.component.Button
import io.jmix.ui.component.Window.HasWorkArea
import io.jmix.ui.component.mainwindow.Drawer
import io.jmix.ui.icon.JmixIcon
import io.jmix.ui.navigation.Route
import io.jmix.ui.screen.*
import org.springframework.beans.factory.annotation.Autowired

@UiController("${normalizedPrefix_underscore}MainScreen")
@UiDescriptor("main-screen.xml")
@Route(path = "main", root = true)
open class MainScreen : Screen(), HasWorkArea {

    @Autowired
    private lateinit var screenTools: ScreenTools

    @Autowired
    private lateinit var workArea: AppWorkArea

    @Autowired
    private lateinit var drawer: Drawer

    @Autowired
    private lateinit var collapseDrawerButton: Button

    override fun getWorkArea(): AppWorkArea = workArea

    @Subscribe("collapseDrawerButton")
    private fun onCollapseDrawerButtonClick(event: Button.ClickEvent) {
        drawer.run {
            toggle()
            if (isCollapsed) {
                collapseDrawerButton.setIconFromSet(JmixIcon.CHEVRON_RIGHT)
            } else {
                collapseDrawerButton.setIconFromSet(JmixIcon.CHEVRON_LEFT)
            }
        }
    }

    @Subscribe
    fun onAfterShow(event: AfterShowEvent?) {
        screenTools.run {
            openDefaultScreen(
                UiControllerUtils.getScreenContext(this@MainScreen).screens
            )
            handleRedirect()
        }
    }
}